File: sample-services/dotnet/src/commands/remove-something.cs
using Sample.Services.Abstractions;

namespace Sample.Services.Commands
{
    public class RemoveSomethingCommand : CommandBase, ICommand
    {
        public const string TYPE = nameof(RemoveSomethingCommand);
        public override string Type => TYPE;
        public int SomeId { get; set; }
        public RemoveSomethingCommand() 
        {
            SomeId = 0;
        }
    }

    public interface IRemoveSomethingCommandHandler : ICommandHandler<RemoveSomethingCommand, int> { }
}

File: sample-services/npm/src/commands/remove-something.ts
import { ICommand } from "../abstractions/command";
import { CommandHandlerBase } from "../abstractions/command-handler";

export interface IRemoveSomethingCommand extends ICommand {
    someId: number;
}

export class RemoveSomethingCommand implements IRemoveSomethingCommand {
    static readonly TYPE = "SampleServices.Commands.RemoveSomethingCommand";
    readonly type: string;
    someId: number;
    constructor(state?: Partial<IRemoveSomethingCommand>) {
        this.someId = 0;
        state && Object.assign(this, JSON.parse(JSON.stringify(state)));
        this.type = RemoveSomethingCommand.TYPE;
    }
}

export interface IRemoveSomethingHandler extends CommandHandlerBase<IRemoveSomethingCommand, number> { }